### DISCUSSION TITLE
JS: Publish as NPM package

### DISCUSSION THREAD

**Discussion by JPustkuchen** [fabianwennink/IconCaptcha-Plugin-Front-End#8] [Ideas]
Just like https://github.com/fabianwennink/IconCaptcha-Plugin-jQuery-PHP/issues/14 for PHP, it would be nice to publish the frontend part as npm package at https://www.npmjs.com/

This is quite simple and would allow to define this as dependency. In the future, splitting these into the different dependency management systems might allow to just require the combination and have the separate libraries updated separately.

For details how to publish to npmjs, see https://docs.npmjs.com/packages-and-modules/contributing-packages-to-the-registry

**Comment by ba77osai** [comment]
@fabianwennink Is there any update on plans to publish this package on NPM?

Publishing on NPM would immensely help us keep the package updated. Currently, we can only pin a specific commit in our build process, as we need to rebundle the library with Webpack and can't use the published server assets.

I believe this would also enhance usability for others using frontend frameworks.

**Reply by fabianwennink** [reply]
Hi @ba77osai! I'm sorry for my delayed response. Sadly I haven't had much time to work on the project recently. Yes, releasing the IconCaptcha widget script on NPM is something I still want to do. I haven't had any experience with this though, so this will be something I need to play around with first. But I'm sure this is not too difficult.

> and can't use the published server assets

Are you referring to the icons here? If so, they are meant to be served by the server requests, so these won't be included in the NPM package.

**Reply by ba77osai** [reply]
Sorry for the missunderstanding, I mean the JS Code of course and not the icons :)

**Reply by PhenX** [reply]
Hello, @fabianwennink I also think publishing on NPM would be a huge plus for this project, if you need help to do it, ideally via GitHub Actions, don't hesitate to ask for help here.
